docs/sdks/client-sdks/javascript/bandits.mdx

```diff
@@ -0,0 +1,294 @@
+---
+title: Contextual Bandits
+sidebar_position: 5
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import ApiOptionRef from '@site/src/components/ApiOptionRef';
+
+## Usage with Contextual Multi-Armed Bandits
+
+Eppo also supports contextual multi-armed bandits. You can read more about them in the [high-level documentation](/contextual-bandits).
+Bandit flag configuration--including setting up the flag key, status quo variation, bandit variation, and targeting rules--are configured within
+the Eppo application. However, available actions are supplied to the SDK in the code when querying the bandit.
+
+To leverage bandits using the JavaScript Client SDK, there are two additional steps over regular feature flags:
+1. Add a bandit action logger to the SDK client instance
+2. Query the bandit for an action
+
+### Define a bandit assignment logger
+
+In order for the bandit to learn an optimized policy, we need to capture and log the bandit's actions.
+This requires defining a bandit logger in addition to an assignment logger.
+```ts
+// Import Eppo's logger interfaces and client initializer
+import { IAssignmentLogger, IBanditLogger, init } from "@eppo/js-client-sdk";
+
+// Define an assignment logger for recording variation assignments
+const assignmentLogger: IAssignmentLogger = {
+  logAssignment(assignment: IAssignmentEvent) {
+    console.log("TODO: save assignment information to data warehouse", assignment);
+  }
+};
+
+// Define a bandit logger for recording bandit action assignments
+const banditLogger: IBanditLogger = {
+  logBanditAction (banditEvent: IBanditEvent) {
+    console.log("TODO: save bandit action information to the data warehouse, ensuring the column names are as expected", banditEvent);
+  }
+};
+
+// Initialize the SDK with both loggers provided
+await init({
+  apiKey: "<SDK_KEY>",
+  assignmentLogger,
+  banditLogger
+});
+```
+
+The SDK will invoke the `logBanditAction()` function with an `IBanditEvent` object that contains the following fields:
+
+<ApiOptionRef 
```

Comment:
I'm not sure if this API Option Ref is better than a table for describing object/data schemas. The idea of a "default" doesn't really apply for instance

Reply:
For whatever reason, I thought the example values were default values. I still don't want to use a table here if we can help it because of the small viewport issues tables end up having. Let me revert this for now. I think a slight variation on the API ref for schemas should take less than a half hour to put together.